x/emissions/keeper/keeper.go

```diff
@@ -1139,44 +1161,205 @@ func (k *Keeper) SetDelegateStakeUponReputer(ctx context.Context, topicId TopicI
-	key := collections.Join3(topicId, reputer, delegator)
-	return k.delegateStakeRemoval.Get(ctx, key)
+// get info about a removal
+func (k Keeper) GetStakeRemoval(
+	ctx context.Context,
+	BlockHeight int64,
+	topicId TopicId,
+	reputer ActorId,
+) (types.StakeRemovalInfo, error) {
+	return k.stakeRemovalsByBlock.Get(ctx, collections.Join3(BlockHeight, topicId, reputer))
+}
+
+// get a list of stake removals for this block
+func (k *Keeper) GetStakeRemovalsForBlock(
+	ctx context.Context,
+	blockHeight BlockHeight,
+) ([]types.StakeRemovalInfo, error) {
+	ret := make([]types.StakeRemovalInfo, 0)
+	rng := collections.NewPrefixedTripleRange[BlockHeight, TopicId, ActorId](blockHeight)
+	iter, err := k.stakeRemovalsByBlock.Iterate(ctx, rng)
+	if err != nil {
+		return ret, err
+	}
+	for ; iter.Valid(); iter.Next() {
+		val, err := iter.Value()
+		if err != nil {
+			return ret, err
+		}
+		ret = append(ret, val)
+	}
+	return ret, nil
+}
+
+// get the first found stake removal for a reputer and topicId or err not found if not found
+func (k *Keeper) GetStakeRemovalForReputerAndTopicId(
+	ctx sdk.Context,
+	reputer string,
+	topicId uint64,
+) (removal types.StakeRemovalInfo, found bool, err error) {
+	rng := collections.NewSuperPrefixedTripleRange[ActorId, TopicId, BlockHeight](reputer, topicId)
+	iter, err := k.stakeRemovalsByActor.Iterate(ctx, rng)
+	if err != nil {
+		return types.StakeRemovalInfo{}, false, err
+	}
+	keys, err := iter.Keys()
+	if err != nil {
+		return types.StakeRemovalInfo{}, false, err
+	}
+	keysLen := len(keys)
+	if keysLen == 0 {
+		return types.StakeRemovalInfo{}, false, nil
+	} else {
+		if keysLen > 1 {
+			ctx.Logger().Warn("Invariant failure! More than one stake removal found for reputer and topicId")
+		}
+		key := keys[0]
+		forwardKey := collections.Join3(key.K3(), topicId, reputer)
+		ret, err := k.stakeRemovalsByBlock.Get(ctx, forwardKey)
+		if err != nil {
+			return types.StakeRemovalInfo{}, false, err
+		}
+		return ret, true, nil
+	}
 }
 
 // For a given address, adds their stake removal information to the removal queue for delay waiting
 // The topic used will be the topic set in the `removalInfo`
 // This completely overrides the existing stake removal
-func (k *Keeper) SetDelegateStakeRemoval(ctx context.Context, removalInfo types.DelegateStakePlacement) error {
-	key := collections.Join3(removalInfo.TopicId, removalInfo.Reputer, removalInfo.Delegator)
-	return k.delegateStakeRemoval.Set(ctx, key, removalInfo)
+func (k *Keeper) SetDelegateStakeRemoval(ctx context.Context, removalInfo types.DelegateStakeRemovalInfo) error {
+	byBlockKey := Join4(removalInfo.BlockRemovalCompleted, removalInfo.TopicId, removalInfo.Delegator, removalInfo.Reputer)
+	err := k.delegateStakeRemovalsByBlock.Set(ctx, byBlockKey, removalInfo)
+	if err != nil {
+		return err
+	}
+	byActorKey := Join4(removalInfo.Delegator, removalInfo.Reputer, removalInfo.TopicId, removalInfo.BlockRemovalCompleted)
```

Comment:
in the other places you had the keys in exact reverse order for block and actor key, that is not the case here. Do you want it to be?